six/modules/c++/cphd/tests/test_compare_cphd.cpp

```diff
@@ -0,0 +1,270 @@
+                     const mem::ScopedArray<sys::ubyte>& data2,
+                     size_t size)
+{
+    for (size_t ii = 0; ii < size; ++ii)
+    {
+        if (data1[ii] != data2[ii])
+        {
+            std::cout <<  "Support data has differing data starting at"
+                      << " index " << ii << "\n";
+            return false;
+        }
+    }
+    return true;
+}
+
+bool compareWideband(cphd::CPHDReader& reader1,
+                     cphd::CPHDReader& reader2,
+                     size_t channelsToProcess,
+                     size_t numThreads)
+{
+    bool dataMatches = true;
+
+    const cphd::Wideband& wideband1 = reader1.getWideband();
+    const cphd::Wideband& wideband2 = reader2.getWideband();
+
+    mem::ScopedArray<sys::ubyte> cphdData1;
+    mem::ScopedArray<sys::ubyte> cphdData2;
+
+    for (size_t ii = 0; ii < channelsToProcess; ++ii)
+    {
+        const types::RowCol<size_t> dims1(
+                reader1.getMetadata().data.getNumVectors(ii),
+                reader1.getMetadata().data.getNumSamples(ii));
+        const types::RowCol<size_t> dims2(
+                reader2.getMetadata().data.getNumVectors(ii),
+                reader2.getMetadata().data.getNumSamples(ii));
+
+        if (dims1 == dims2)
+        {
+            wideband1.read(ii,
+                           0, cphd::Wideband::ALL, 0,
+                           cphd::Wideband::ALL,
+                           numThreads, cphdData1);
+
+            wideband2.read(ii,
+                           0, cphd::Wideband::ALL, 0,
+                           cphd::Wideband::ALL,
+                           numThreads, cphdData2);
+
+            switch (reader1.getMetadata().data.getSignalFormat())
+            {
+            case cphd::SampleType::RE08I_IM08I:
+                if (!compareCPHDData<std::complex<sys::Int8_T> >(
+                        cphdData1.get(),
+                        cphdData2.get(),
+                        dims1.area(),
+                        ii))
+                {
+                    dataMatches = false;
+                }
+                break;
+            case cphd::SampleType::RE16I_IM16I:
+                if (!compareCPHDData<std::complex<sys::Int16_T> >(
+                        cphdData1.get(),
+                        cphdData2.get(),
+                        dims1.area(),
+                        ii))
+                {
+                    dataMatches = false;
+                }
+                break;
+            case cphd::SampleType::RE32F_IM32F:
+                if (!compareCPHDData<std::complex<float> >(
+                        cphdData1.get(),
+                        cphdData2.get(),
+                        dims1.area(),
+                        ii))
+                {
+                    dataMatches = false;
+                }
+                break;
+            }
+        }
+        else
+        {
+            std::cout << "Data at channel " << ii
```

Comment:
Use `cerr` for failure messages. (I forget the behavior exactly, but the test runner handles stdout and stderr differently in some reasonable manner)